### DISCUSSION TITLE
Renovate schedule issues (prevent bot from pushing more updates to existing PR)

### DISCUSSION THREAD

**Discussion by noahflk** [renovatebot/renovate#15744] [Request Help]
## Config

```json
{
  "extends": ["config:base"],
  "stabilityDays": 7,
  "prCreation": "not-pending",
  "timezone": "Europe/Berlin",
  "schedule": ["before 6am on Monday"],
  "packageRules": [
    {
      "matchPackagePatterns": ["*"],
      "matchUpdateTypes": ["minor", "patch"],
      "groupName": "all non-major dependencies",
      "groupSlug": "all-minor-patch",
      "labels": ["dependencies"],
      "automerge": true
    },
    {
      "matchPackagePatterns": ["*"],
      "matchUpdateTypes": ["major"],
      "labels": ["dependencies", "breaking"],
      "automerge": false
    }
  ]
}
```

## Expected behavior
- Every Monday morning I get a PR containing all `minor` and `patch` updates
- The PR will remain open until all package updates in there are at least 7 days old
- At that point the PR will be auto merged

## Actual behavior
New updates get added to the existing PR whenever they come out. It doesn't respect the Monday morning schedule. This means that the PR *never* gets automerged because there are always updates less than 7 days old.

I assume this happens because the schedule dictates PR creation but not when updates get pushed. So what's the best way to achieve my expected behavior? Would adding this property to the group help?

```json
"schedule": ["before 6am on Monday"],
```

**Comment by viceice** [comment]
schedule is for branch creation not PR creation time.

set internal filter  to strict, then only stable time passed updates are added to PR. you can also disable branch update outside schedule to prevent further PR update.

- https://docs.renovatebot.com/configuration-options/#internalchecksfilter
- https://docs.renovatebot.com/configuration-options/#updatenotscheduled
